go/types/indexed_sequences.go

```diff
@@ -4,20 +4,14 @@
 
 package types
 
-import (
-	"sort"
-
-	"github.com/attic-labs/noms/go/d"
-)
+import "github.com/attic-labs/noms/go/d"
 
 type indexedSequence interface {
 	sequence
-	cumulativeNumberOfLeaves(idx int) uint64 // returns the total number of leaf values reachable from this sequence for all sub-trees from 0 to |idx|
 }
 
 type indexedMetaSequence struct {
 	metaSequenceObject
```

Comment:
It would be great to collapse the `indexedMetaSequence` and `orderedMetaSequence` into just the `metaSequence` struct, but `oMS` has `getKey()`.

Which, you could put on `metaSequence`, via putting on `sequence`, which *still* makes sense - keys are a sequence concept, not an ordered sequence concept - *but* they're used for binary searching, which indexed sequences don't support.

And *then* you could just make `metaSequence` a struct, not an interface.

Whew. Thoughts? All I really need now is `cumulativeNumberOfLeaves()` on `sequence`.